File: MobileTrailer/GithubViewController.m

@implementation GithubViewController

- (void)viewDidLoad
{
	[super viewDidLoad];

	WKWebViewConfiguration *webConfiguration = [[WKWebViewConfiguration alloc] init];
	self.webView = [[WKWebView alloc] initWithFrame:self.view.bounds
									  configuration:webConfiguration];
	self.webView.navigationDelegate = self;
	self.webView.scrollView.contentInset = UIEdgeInsetsMake(64, 0, 0, 0);
	self.webView.autoresizingMask = UIViewAutoresizingFlexibleWidth|UIViewAutoresizingFlexibleHeight;
	self.webView.translatesAutoresizingMaskIntoConstraints = YES;
	[self.view addSubview:self.webView];

	[self.webView loadRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:self.pathToLoad]]];
}

- (void)webView:(WKWebView *)webView didFailNavigation:(WKNavigation *)navigation withError:(NSError *)error
{
	[self loadingFailed];
}

- (void)webView:(WKWebView *)webView didFailProvisionalNavigation:(WKNavigation *)navigation withError:(NSError *)error
{
	[self loadingFailed];
}

- (void)loadingFailed
{
	self.webView.hidden = NO;
	[self.spinner stopAnimating];
	self.title = @"Loading Failed";
}

- (void)webView:(WKWebView *)webView didFinishNavigation:(WKNavigation *)navigation
{
	self.webView.hidden = NO;
	[self.spinner stopAnimating];
	self.title = self.webView.title;
}

- (void)webView:(WKWebView *)webView didStartProvisionalNavigation:(WKNavigation *)navigation
{
	self.webView.hidden = YES;
	[self.spinner startAnimating];
}

- (IBAction)ipadDone:(UIBarButtonItem *)sender
{
	[self dismissViewControllerAnimated:YES completion:nil];
}

@end

File: MobileTrailer/GithubViewController.swift
import WebKit

class GithubViewController: UIViewController, WKNavigationDelegate {
	
	@IBOutlet var spinner: UIActivityIndicatorView!
	var pathToLoad: String?
	private var _webView: WKWebView?

	override func viewDidLoad() {
		super.viewDidLoad()

		let webConfiguration = WKWebViewConfiguration()
		_webView = WKWebView(frame: view.bounds, configuration: webConfiguration)
		_webView!.navigationDelegate = self
		_webView!.scrollView.contentInset = UIEdgeInsetsMake(64, 0, 0, 0)
		_webView!.autoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;
		_webView!.setTranslatesAutoresizingMaskIntoConstraints(true)
		view.addSubview(_webView!)
		_webView!.loadRequest(NSURLRequest(URL: NSURL(string: pathToLoad!)!))
	}

	func webView(webView: WKWebView, didFailNavigation navigation: WKNavigation!, withError error: NSError) {
		loadingFailed()
	}

	func webView(webView: WKWebView, didFailProvisionalNavigation navigation: WKNavigation!, withError error: NSError) {
		loadingFailed()
	}

	private func loadingFailed() {
		_webView!.hidden = false
		spinner.stopAnimating()
		title = "Loading Failed"
		UIAlertView(title: "Loading Failed", message: "URL: '"+pathToLoad!+"'", delegate: nil, cancelButtonTitle: "OK").show()
	}

	func webView(webView: WKWebView, didFinishNavigation navigation: WKNavigation!) {
		webView.hidden = false
		spinner.stopAnimating()
		title = webView.title

	}

	@IBAction func doneSelected(sender: UIBarButtonItem) {
		dismissViewControllerAnimated(true, completion: nil)
	}

	func webView(webView: WKWebView, didStartProvisionalNavigation navigation: WKNavigation!) {
		webView.hidden = true
		spinner.startAnimating()
	}
}
